testsuite/tests/asmcomp/polling_insertion.ml

```diff
@@ -0,0 +1,254 @@
+(* TEST
+   modules = "polling.c"
+   compare_programs = "false"
+   * native
+*)
+
+(* This set of tests examine poll insertion behaviour. We do this by requesting
+   and checking the number of minor collections at various points to determine
+   whether a poll was correctly added. There are some subtleties because 
+   [caml_empty_minor_heap] will not increment the minor_collections stat if
+   nothing has been allocated on the minor heap, so we sometimes need to
+   add an allocation before we call [request_minor_gc]. The [minor_gcs]
+   function returns the number of minor collections so far without allocating.
+
+   ignore(Sys.opaque_identity(ref 41)) is used wherever we want to do an
+   allocation in order to use some minor heap in order for the minor
+   collections stat to be incremenets.
+
+   ignore(Sys.opaque_identity(ref 42)) is used wherever we want an allocation
+   for the purposes of testing whether a poll would be elided or not.
+*)
+
+external request_minor_gc : unit -> unit = "request_minor_gc"
+external minor_gcs : unit -> int = "minor_gcs"
+
+(* This function tests that polls are added only to the back edges of loops *)
+let polls_added_to_loops () =
+  let minors_before = minor_gcs () in
+  request_minor_gc ();
+  for a = 0 to 1 do
+    let minors_now = minor_gcs () in
+    if a == 0 then
```

Comment:
These `==` can just be `=` as physical equality isn't relevant to integer comparisons.

Reply:
I see. Fixing!